### PR TITLE
cleaner volume script for extensibility and swayosd integration

### PR DESCRIPTION
**Pull request by keth-real**
# Pull Request

## Description
A different approach from this https://github.com/prasanthrangan/hyprdots/pull/1818.
This way we can adhere to the original configuration without breaking legacy functions. 

The waybar modules,keybindings.conf will not be touched.

Should I add these in ` ./hyprland.conf `  too?  https://github.com/prasanthrangan/hyprdots/blob/f92564a22abd0a673cfdd035b0ef7009d9091c45/Configs/.config/hypr/hyprland.conf#L41
```
exec-once = swayosd-server # enable swayosd service
exec-once = pkexec swayosd-libinput-backend # swayosd service for keyboard input
```

Needs https://github.com/prasanthrangan/hyprdots/pull/1818 to be resolved first. 

Should resolve the ff:

```
./hyprland.conf
 ./custom_app.lst
./restore_cfg.lst
```

Added 
` volumecontrol.sh  -t `  to toggle output device from here https://discord.com/channels/1200448076620501063/1283781653310734418/1283851047554781324

Fixes: 
The old script do not have the ability to toggle mute a player. This impl should be useful. 

## Type of change

Please put an `x` in the boxes that apply:

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [x] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [x] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [x] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added a changelog entry.
- [x] I have added necessary comments/documentation to my code.
- [x] I have added tests to cover my changes.
- [x] I have tested my code locally and it works as expected.
- [x] All new and existing tests passed.

## Screenshots

(if appropriate)

## Additional context

This PR is not yet complete and will be discussed here https://github.com/prasanthrangan/hyprdots/pull/1818

### CODE REVIEW

**Comment by kRHYME7** [comment]
damn, another 1. 

@rubinV @TheyCallMeSecond

Can you alsoe please check this one? This one has a good approach though, without changing the keybindings.conf.

**Comment by kRHYME7** [comment]
Will draft first. Updraft when ready. Thanks!

File: Configs/.local/share/bin/brightnesscontrol.sh
Comment (kRHYME7) [inline Configs/.local/share/bin/brightnesscontrol.sh:1]: Cool, Please make the script consistent with the volumescript. Thanks!

**Review by kRHYME7 — CHANGES_REQUESTED** [review]

**Comment by keth-real** [comment]
Ready for testing @rubiin  @kRHYME7

**Comment by rubiin** [comment]
When you say extensible is it backward compatible ?

**Comment by keth-real** [comment]
> When you say extensible is it backward compatible ?

The existing version supports the older script, which can be confusing and makes it difficult for other contributors to understand. Additionally, I have fixed the -p player option and implemented fixes for the toggle mute for -p. The older script relies on the action_function format, where the function is from another variable, leading to unnecessary confusion.

**Comment by rubiin** [comment]
Cool. Will give it a try tomorrow and get back to you.

**Comment by kRHYME7** [comment]
```
# exec-once = pkexec swayosd-libinput-backend | - # swayosd service for keyboard input. Need to add the '| -' for now so the daemon won't exit 
```

Note we can run the daemon on user level but we need to trick this deamon as this is intended to run using systemctl
We  can trick it by piping it somewhere  

we can use | something 
or we can  use && or || or any ways we can execute a subshell while invokinh swayosd-libinput-backend

This way swayosd-libinput stays on the user level.


I'm just amaze with this bug.

**Comment by kRHYME7** [comment]
@rubiin  This works on my end. Also, swayosd is snappier than other OSD I tried.

**Comment by rubiin** [comment]
> @rubiin This works on my end. Also, swayosd is snappier than other OSD I tried.

works on mine too

**Comment by kRHYME7** [comment]
This works as intended and shouldn't give problems if users do not have Swayosd.

We can Merge this first then refine the other PR? How about that?

**Comment by rubiin** [comment]
Lets get this merged

**Review by kRHYME7 — APPROVED** [review]
💗

**Pull request merged**